### ISSUE TITLE
Automatically recover from recoverable errors

### ISSUE THREAD

**Issue by jospoortvliet** [nextcloud/desktop#1322]
Right now, when a download fails, the client will put an error in the error list and try again. Often, things are fine then - and the client continues, but the error is in the error list, disturbing the user needlessly. I suggest to wait until a retry or two has failed before an error is shown. In general, the client is quite 'chatty', a bit more quiet would probably be better...

**Comment by Croydon** [comment]
Yes please. Quite often, I restart the client just to get rid of old error messages that already resolved themselves. but did not disappear automatically, nor do they allow to dismiss them manually.

**Comment by Rello** [comment]
The client received a lot of optimisations on the error handling and self discovery. if the error still exists, please let us know
